source/request-as-event-emitter.js

```diff
@@ -32,7 +32,12 @@ module.exports = options => {
 			return;
 		}
 
-		let fn = options.protocol === 'https:' ? https : http;
+		let fn;
+		if (options.fn) {
```

Comment:
Let's name it `options._requestFunction` so it's clear that it's readable and clear that not something we officially support.